shared/engine/index.js

```diff
@@ -410,6 +414,7 @@ class Engine {
 
   reset () {
     resetClient()
+    this._setupClient()
```

Comment:
Doesn't look it works on mobile. Even just doing `resetClient()` breaks future rpc requests.

I think we should hold off on mobile hmr engine until you finish your changes.